### PR TITLE
Add OpenShift apiserver ServiceMonitor

### PR DESCRIPTION
**Pull request by brancz**
These changes add the OpenShift apiserver targets to be scraped. Works like a charm on an AWS cluster for me:

screenshot from 2018-12-04 12-34-04

@s-urbaniak @squat @metalmatze

### CODE REVIEW

File: jsonnet/prometheus.jsonnet [hunk 1]
```diff
@@ -269,6 +269,34 @@ local namespacesRole =
         },
       },
 
+    serviceMonitorOpenShiftApiserver:
```
Comment (s-urbaniak) [inline jsonnet/prometheus.jsonnet]: nit: we could make this a function, as more and more service monitors are popping up with similar/same settings.

**Comment by squat** [comment]
> nit: we could make this a function, as more and more service monitors are popping up with similar/same settings.
 Resolve conversation

I think that the ideal path forward here would be to develop a prometheus-operator library of mixins like ksonnet so we can easily and reliably create new prometheus-operator owned objects

File: assets/prometheus-k8s/service-monitor-open-shift-apiserver.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+    interval: 30s
+    port: https
+    scheme: https
+    tlsConfig:
+      caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
+      serverName: api.openshift-apiserver.svc
+  namespaceSelector:
+    matchNames:
+    - openshift-apiserver
```
[the remark this answers is not in the document]
Comment (brancz) [reply assets/prometheus-k8s/service-monitor-open-shift-apiserver.yaml]: there is only a single endpoint, and no selector means it selects all

**Comment by brancz** [comment]
> I think that the ideal path forward here would be to develop a prometheus-operator library of mixins like ksonnet so we can easily and reliably create new prometheus-operator owned objects

Exactly my thought.

**Pull request merged**